samples/net/mqtt/src/modules/error/Kconfig.error

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig MQTT_SAMPLE_ERROR
+	bool "Error"
+	default y
+
+if MQTT_SAMPLE_ERROR
+
+config MQTT_SAMPLE_ERROR_REBOOT_ON_FATAL
+	bool "Reboot on fatal error"
```

Comment:
It should possibly be noted somewhere here that this is about app faults, not system faults per se. They will still reboot or not based on `CONFIG_RESET_ON_FATAL_ERROR`